Fakebook.Posts/Fakebook.Posts.UnitTests/Controllers/PostControllerTest.cs

```diff
@@ -204,15 +211,63 @@ public async Task PostAsyncComment_InvalidPost_BadRequest()
                 });
             }).CreateClient();
 
+            NewCommentDTO newComment = new() { Content = "Valid Content" };
+
             client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Test");
 
-            StringContent stringContent = new(System.Text.Json.JsonSerializer.Serialize(comment), Encoding.UTF8, "application/json");
+            StringContent stringContent = new(System.Text.Json.JsonSerializer.Serialize(newComment), Encoding.UTF8, "application/json");
 
             // Act
             var response = await client.PostAsync("api/comments", stringContent);
 
             // Assert
             Assert.Equal(System.Net.HttpStatusCode.BadRequest, response.StatusCode);
         }
+
+        /// <summary>
+        /// Tests the PostsController clases ability to reject an invalid DTO.
+        /// </summary>
+        [Fact]
+        public async Task PostAsync_InvalidPostDTO_BadRequest()
+        {
+            // Arrange
+            Mock<IPostsRepository> mockRepo = new();
+            Mock<IFollowsRepository> mockFollowRepo = new();
+            Mock<IBlobService> mockBlobService = new();
+            mockRepo.Setup(r => r.AddAsync(It.IsAny<Post>()))
+                .Throws(new DbUpdateException());
+
+            List<Comment> comments = new();
+            var date = DateTime.Now;
+            Post post = new("test.user@email.com", "test content")
+            {
+                Id = 1,
+                Comments = comments,
+                Picture = "picture",
+                CreatedAt = date
+            };
+
+            var client = _factory.WithWebHostBuilder(builder =>
+            {
+                builder.ConfigureTestServices(services =>
+                {
+                    services.AddScoped(sp => mockRepo.Object);
+                    services.AddScoped(sp => mockFollowRepo.Object);
+                    services.AddTransient(sp => mockBlobService.Object);
+                    services.AddAuthentication("Test")
+                        .AddScheme<AuthenticationSchemeOptions, TestAuthHandler>("Test", options => { });
+                });
+            }).CreateClient();
+
+            client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Test");
+
+            StringContent stringContent = new(System.Text.Json.JsonSerializer.Serialize(post), Encoding.UTF8, "application/json");
+
+            // Act
+            var response = await client.PostAsync("api/posts", stringContent);
+
+            // Assert
+            Assert.Equal(System.Net.HttpStatusCode.BadRequest, response.StatusCode);
```

Comment:
why not use `using` directives for all these namespaces

Reply:
This method is actually not supposed to be here, it was the start of me thinking how I was going to test data annotation specific behaviour.  I plan on doing that in the current test methods now though with addition asserts and responses.  That function has been removed, I'll consider this for the existing methods though, because that is how it's being done currently and wasn't something I chose